### ISSUE TITLE
Installation problems and problems with TBD001

### ISSUE THREAD

**Issue by bclmary** [tWido/TMC#1]
Hy,

Here are some issues I encountered during installation on a debian Sid.
* The 3d line of the Makefile didn't work on my shell, I had to blank the variable PATH_PREFIX.
* The Linker awaits ftd2xx to be copied in system library (at least, in /usr/local/lib/). This is specified in the ReadMe of the driver, but not in your documentation. An alternative could be to set the linker's library path differently.

What follows may be specific with Thorlabs TBD001 (but I don't think so...):
* In cmd.cpp, some device_calls functions are switched in conditional blocs of several command functions: StartRelMoveC, StartAbsMoveC (and perhaps StartJogMoveC, I didn't used this one). This make these commands unusable with TBD001 at least.
For example, in StartRelMoveC, conditional bloc:
```
    if (dist_spec) ret = device_calls::StartSetRelativeMove(0x50, index);
    else ret = device_calls::StartRelativeMove(dist, 0x50, index); 
```
should be:
```
    if (dist_spec) ret = device_calls::StartRelativeMove(dist, 0x50, index); 
    else ret = device_calls::StartSetRelativeMove(0x50, index);
```
* TBD001 model is not set in 'functionsSet' of init.cpp (line 303).

These are more subjective:
* A tab/spaces at the beginning of each return of a command may enhance readability. For example:
```
relmove -h
    Start relative move. Distance can be either specified or previously set by relmovep command.
    -i NUMBER           index of motor to start move, default is 1
    -d VALUE            distance to move, if not specified parameter stored in device is used
relmove -d ...
```
* The very first message before prompt should be preceded by a blank line (I stupidly didn't see it the very first time !). By the way, It may be useful to write that '-h' option is available for every command (took me some time to figure that out).
This is on line 1625 of file cmd.cpp. My proposition :
`printf("\nAwaiting commands. Type \"help\" to display available commands.\n(tip: \"-h\" option is available for all commands to obtain specific help)\n");`

Great job anyway :-D

**Comment by tWido** [comment]
> The 3d line of the Makefile didn't work on my shell, I had to blank the variable 

What shell do you use?

> The Linker awaits ftd2xx to be copied in system library (at least, in /usr/local/lib/). This is specified in the ReadMe of the driver, but not in your documentation. An alternative could be to set the linker's library path differently.

This is the result of blanking the PATH_PREFIX variable. File ftd2xx.so is linked from unpacked driver directory tree.

> What follows may be specific with Thorlabs TBD001 (but I don't think so...):

Yeah. Those are surely bugs. I'll fix them.

> These are more subjective

Both sound good. I'll do it


Thanks for feedback. :+1:  I will try to spend more time on this project.
 Feel free to contribute by forking code and creating PR. And of course, filling out issues.

**Comment by bclmary** [comment]
> This is the result of blanking the PATH_PREFIX variable. File ftd2xx.so is linked from unpacked driver directory tree.

Yes ! I figure that out few time after posting. This issue appears because... suspense... I had a folder name with a space in the name !

> Feel free to contribute by forking code and creating PR.

I'm new in contributing on codes. I will post something just for the fun of it ;-)

**Comment by tWido** [comment]
Solved #6
